src/pact/v3/error.py

```diff
@@ -0,0 +1,831 @@
+        return self._description
+
+    @property
+    def error(self) -> Exception:
+        """
+        Error that occurred during the verification of the interaction.
+        """
+        return self._error
+
+
+class PactVerificationError(PactError):
+    """
+    Exception raised due to errors in the verification of a Pact.
+
+    This is raised when performing manual verification of the Pact through the
+    [`verify`][pact.v3.Pact.verify] method:
+
+    ```python
+    pact = Pact("consumer", "provider")
+    # Define interactions...
+    try:
+        pact.verify(handler, kind="Async")
+    except PactVerificationError as e:
+        print(e.errors)
+    ```
+
+    All of the errors that occurred during the verification of all of the
+    interactions are stored in the `errors` attribute.
+
+    This is different from the [`MismatchesError`][pact.v3.MismatchesError]
+    which is raised when there are mismatches detected by the mock server.
+    """
+
+    def __init__(self, errors: list[InteractionVerificationError]) -> None:
+        """
+        Initialise a new PactVerificationError.
+
+        Args:
+            errors:
+                Errors that occurred during the verification of the Pact.
+        """
+        super().__init__(f"Error verifying Pact (count: {len(errors)})")
+        self._errors = errors
+
+    @property
+    def errors(self) -> list[InteractionVerificationError]:
+        """
+        Errors that occurred during the verification of the Pact.
+        """
+        return self._errors
+
+
+class Mismatch(ABC):  # noqa: B024
+    """
+    A mismatch between the Pact contract and the actual interaction.
+
+    See
+    https://github.com/pact-foundation/pact-reference/blob/f5ddf3d353149ae0fb539a1616eeb8544509fdfc/rust/pact_matching/src/lib.rs#L880
+    for the underlying source of the data.
+    """
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Mismatch:  # noqa: PLR0911
+        """
+        Create a new Mismatch from a dictionary.
+
+        Args:
+            data:
+                Data for the mismatch.
+
+        Returns:
+            A new Mismatch object.
+        """
+        if mismatch_type := data.pop("type"):
+            if mismatch_type == "MethodMismatch":
+                return MethodMismatch(**data)
+            if mismatch_type == "PathMismatch":
+                return PathMismatch(**data)
+            if mismatch_type == "StatusMismatch":
+                return StatusMismatch(**data)
+            if mismatch_type == "QueryMismatch":
+                return QueryMismatch(**data)
+            if mismatch_type == "HeaderMismatch":
+                return HeaderMismatch(**data)
+            if mismatch_type == "BodyTypeMismatch":
+                return BodyTypeMismatch(**data)
+            if mismatch_type == "BodyMismatch":
+                return BodyMismatch(**data)
+            if mismatch_type == "MetadataMismatch":
+                return MetadataMismatch(**data)
```

Comment:
I could not provoke any of those types.
All types I could provoke so far (in consumer tests) are Kebab Case
`request-not-found` or `missing-request`
so they will always end up as `GenericMismatch`. Is that intended?